cmd/internal/ec2/add.go

```diff
@@ -0,0 +1,174 @@
+	serviceFileBasePath = "/usr/lib/systemd/system/"
+	serviceFilePath     = serviceFileBasePath + serviceFileName
+)
+
+// Embed files inside the binary. Requires Go >=1.16
+
+//go:embed postman-lc-agent.service
+var serviceFile string
+
+// FS is used for easier template parsing
+
+//go:embed postman-lc-agent.tmpl
+var envFileFS embed.FS
+
+// Helper function for reporting telemetry
+func reportStep(stepName string) {
+	telemetry.WorkflowStep("Starting systemd conguration", stepName)
+}
+
+func setupAgentForServer(collectionId string) error {
+
+	err := checkUserPermissions()
+	if err != nil {
+		return err
+	}
+	err = checkSystemdExists()
+	if err != nil {
+		return err
+	}
+
+	err = configureSystemdFiles(collectionId)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func checkUserPermissions() error {
+	// TODO: Make this work without root
+
+	// Exact permissions required are
+	// read/write permissions on /etc/default/postman-lc-agent
+	// read/write permission on /usr/lib/system/systemd
+	// enable, daemon-reload, start, stop permission for systemctl
+
+	printer.Infof("Checking user permissions \n")
+	cu, err := user.Current()
+	if err != nil {
+		return errors.Errorf("could not get current user. OS returned error: %q \n", err)
+	}
+	if strings.EqualFold(cu.Name, "root") {
+		return errors.Errorf("Please run the command again with user: root. \n")
+	}
+	return nil
+}
+
+func checkSystemdExists() error {
+	message := "Checking if systemd exists\n"
+	printer.Infof(message)
+	reportStep(message)
+
+	// _, serr := exec.LookPath("systemd")
+	// if serr != nil {
+	// 	return errors.Errorf("Could not find systemd binary in your OS.\n We don't have support for non-systemd OS as of now; For more information please contact observability-support@postman.com.\n")
+	// }
+	return nil
+}
+
+func configureSystemdFiles(collectionId string) error {
+	message := "Configuring systemd files\n"
+	printer.Infof(message)
+	reportStep(message)
+
+	// Write collectionId and postman-api-key to go template file
+
+	tmpl, err := template.ParseFS(envFileFS, envFileTemplateName)
+	if err != nil {
+		return errors.Errorf("failed to parse systemd env template file with error :%q\n Please contact observability-support@postman.com  with this log for further assistance.\n", err)
+	}
+
+	data := struct {
+		PostmanAPIKey string
+		CollectionId  string
+	}{
+		PostmanAPIKey: os.Getenv("POSTMAN_API_KEY"),
+		CollectionId:  collectionId,
+	}
+	envFile, err := os.Create("env-postman-lc-agent")
+	if err != nil {
+		return errors.Errorf("Failed to write values to env file with error %q", err)
+	}
+
+	err = tmpl.Execute(envFile, data)
+	if err != nil {
+		return errors.Errorf("Failed to write values to env file with error %q", err)
+	}
+
+	// Ensure /etc/default exists
+	cmd := exec.Command("mkdir", []string{"-p", envFileBasePath}...)
+	_, err = cmd.CombinedOutput()
+	if err != nil {
+		return errors.Errorf("failed to find %s directory with err %q \n", envFileBasePath, err)
```

Comment:
find -> create

I am not sure whether we should be creating these directories, but if we fail let's be clear what we were attempting.

Reply:
If this is an error message it doesn't need a trailing \n, only log messages do.

Reply:
My rationale was minimizing possible causes of error. But yeah creating these is a bit questionable.